server/src/main/java/org/opensearch/cluster/metadata/Metadata.java

```diff
@@ -1325,6 +1329,36 @@ public Builder removeDataStream(String name) {
             return this;
         }
 
+        private Map<String, View> getViews() {
+            return Optional.ofNullable(customs.get(ViewMetadata.TYPE))
+                .map(o -> (ViewMetadata) o)
+                .map(vmd -> vmd.views())
+                .orElse(new HashMap<>());
+        }
+
+        public View view(final String viewName) {
+            return getViews().get(viewName);
+        }
+
+        public Builder views(final Map<String, View> views) {
+            this.customs.put(ViewMetadata.TYPE, new ViewMetadata(views));
+            return this;
+        }
+
+        public Builder put(final View view) {
+            Objects.requireNonNull(view, "view cannot be null");
+            final var replacementViews = new HashMap<>(getViews());
+            replacementViews.put(view.getName(), view);
+            return views(replacementViews);
+        }
+
+        public Builder removeView(final String viewName) {
+            Objects.requireNonNull(viewName, "viewName cannot be null");
+            final var replacementViews = new HashMap<>(getViews());
```

Comment:
Can we do `var` yet? I thought we are still on a lower language level.

Reply:
I'll double check the 2.x line and convert if I can't find them, they are so much easier to read.

Reply:
2.x has usage of var - when java 8 support was dropped for 2.x this became possible.  I'm not going to rewrite this PR to only use var, but I'd say we shouldn't shy aware from it.  Or you can add a checkstyle/spotless rule to keep it in/out of the codebase

https://github.com/opensearch-project/OpenSearch/blob/644416f056029cb0b74908014acadc6c02b3c215/server/src/main/java/org/opensearch/common/FieldMemoryStats.java#L76-L79